quinn-proto/src/connection/qlog.rs

```diff
@@ -111,63 +154,285 @@ impl QlogSink {
                 }),
             };
 
-            stream.emit_event(orig_rem_cid, EventData::PacketLost(event), now);
+            stream.emit_event(initial_dst_cid, EventData::PacketLost(event), now);
         }
     }
 
-    pub(super) fn emit_packet_sent(
-        &self,
-        pn: u64,
-        len: usize,
-        space: SpaceId,
-        is_0rtt: bool,
-        now: Instant,
-        orig_rem_cid: ConnectionId,
-    ) {
+    pub(super) fn emit_packet_sent(&self, conn: &Connection, packet: QlogSentPacket, now: Instant) {
         #[cfg(feature = "qlog")]
         {
             let Some(stream) = self.stream.as_ref() else {
                 return;
             };
-
-            let event = PacketSent {
-                header: PacketHeader {
-                    packet_number: Some(pn),
-                    packet_type: packet_type(space, is_0rtt),
-                    length: Some(len as u16),
-                    ..Default::default()
-                },
-                ..Default::default()
-            };
-
-            stream.emit_event(orig_rem_cid, EventData::PacketSent(event), now);
+            stream.emit_event(
+                conn.initial_dst_cid,
+                EventData::PacketSent(packet.inner),
+                now,
+            );
         }
     }
 
     pub(super) fn emit_packet_received(
         &self,
-        pn: u64,
-        space: SpaceId,
-        is_0rtt: bool,
+        conn: &Connection,
+        packet: QlogRecvPacket,
         now: Instant,
-        orig_rem_cid: ConnectionId,
     ) {
         #[cfg(feature = "qlog")]
         {
             let Some(stream) = self.stream.as_ref() else {
                 return;
             };
+            let mut packet = packet;
+            packet.emit_padding();
+            let event = packet.inner;
+            stream.emit_event(conn.initial_dst_cid, EventData::PacketReceived(event), now);
+        }
+    }
+}
 
-            let event = PacketReceived {
-                header: PacketHeader {
-                    packet_number: Some(pn),
-                    packet_type: packet_type(space, is_0rtt),
-                    ..Default::default()
-                },
-                ..Default::default()
-            };
+#[derive(Default)]
+pub(crate) struct QlogSentPacket {
+    #[cfg(feature = "qlog")]
+    inner: PacketSent,
+}
 
-            stream.emit_event(orig_rem_cid, EventData::PacketReceived(event), now);
+#[cfg(feature = "qlog")]
```

Comment:
I think the philosophy of this module is to keep all/most the cfgs inside the function bodies. That way the call sites are not littered with cfgs. And the compiler is expected to optimise out the noop function calls.  So you probably should follow the same pattern for these new impls?

Reply:
Yes, maybe. For QlogRecvPacket it's straightforward, will change. I kept the cfg initially for `QlogSentPacket::frame`, which takes a `qlog::QuicFrame`. Constructing this from the raw data sometimes involves calculations, so to make sure to not do these when qlog is disabled I made the fn be configured out in that case. Maybe I just keep the cfg for that one only.